crates/solver/src/solver/oneinch_solver.rs

```diff
@@ -74,16 +80,50 @@ impl OneInchSolver {
             allowance_fetcher: Box::new(AllowanceManager::new(web3, settlement_address)),
             protocol_cache: ProtocolCache::default(),
             oneinch_slippage_bps,
+            max_slippage_in_wei,
         })
     }
 }
 
 impl OneInchSolver {
-    /// Settles a single sell order against a 1Inch swap using the specified protocols.
-    async fn settle_order_with_protocols(
+    /// Computes the max slippage we are willing to use for a given trade to limit the absolute
+    /// slippage to a configured upper limit in terms of wei. Because 1Inch keeps positive
+    /// slippage, always applying a default slippage would otherwise become very costly for huge
+    /// orders.
+    fn compute_max_slippage(
+        external_buy_token_price_in_wei: &BigRational,
+        buy_amount: &U256,
+        default_slippage_bps: u32,
+        max_slippage_in_wei: &U256,
+    ) -> Result<Slippage> {
+        let max_slippage_in_buy_token =
+            max_slippage_in_wei.to_big_rational() / external_buy_token_price_in_wei;
+
+        let max_slippage_percent_respecting_wei_limit =
+            max_slippage_in_buy_token / buy_amount.to_big_rational();
```

Comment:
The name is misleading because this isn't a percentage. It is a normal ratio. Could clarify that by naming variables max_absolute_slippage_buy_token and max_relative_slippage_wei_limit or something like that.